juniper_official/lldp/lldp-session-state.rule

```diff
@@ -0,0 +1,173 @@
+                description "LLDP neighbor state";
+            }
+            /*
+             * Anomaly detection logic.
+             */
+            trigger lldp-session-state {
+                synopsis "LLDP state KPI";
+                description "Sets health based on lldp session state changes and notify anomaly.";
+                frequency 1offset;
+                /*
+                 * Sets color to green when LLDP neighbor state is true.
+                 */
+                term is-lldp-session-up {
+                    when {
+                        matches-with "$neighbor-state" true;
+                        matches-with "$if-lldp-state" true;
+                    }
+                    then {
+                        status {
+                            color green;
+                            message "Interface $interface-name lldp session is up";
+                        }
+                    }
+                }
+                /*
+                 * Defaults color to red and notify anomaly.
+                 */
+                term lldp-session-down {
+                    then {
+                        status {
+                            color red;
+                            message "Interface-$interface-name lldp session down";
+                        }
+                    }
+                }
+            }
+            /*
+             * Variables with default values. Default values can be changed
+             * while deploying playbook instance.
+             */
+            variable if-name {
+                value ".*";
+                type string;
+                description "Interface names to monitor, regular expression, eg 'ge-.*'";
+           }
+        }
+        rule get-lldp-state {
+            keys interface-name;
+            synopsis "LLDP neighbor state collector";
+            description "Collects lldp session state periodically";
+            sensor lldp-sensor {
+                synopsis "LLDP open-config sensor definition";
+                description "/lldp open-config sensor to collect telemetry data from network device";
+                open-config {
+                    sensor-name /lldp/;
+                    frequency 55s;
+                }
+            }
+            field interface-name {
+                sensor lldp-sensor {
+                    where "/lldp/interfaces/interface/@name =~ /{{if-name}}/";
+                    path "/lldp/interfaces/interface/@name";
+                }
+                type string;
+                description "Interfaces to be monitored";
+            }
+            field neighbor-capability-state {
+                sensor lldp-sensor {
+                    path /lldp/interfaces/interface/neighbors/neighbor/capabilities/capability/state/enabled;
+                }
+                type string;
+                description "Remote LLDP neighbor state";
+            }
+            variable if-name {
+                value .*;
```

Comment:
There is no trigger in this rule. Why is this being used?

Reply:
Device rule will be used in network rule